joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/callgraph/StaticCallLinker.scala

```diff
@@ -1,64 +1,51 @@
 package io.joern.x2cpg.passes.callgraph
 
+import io.joern.x2cpg.utils.LinkingUtil
 import io.shiftleft.codepropertygraph.Cpg
-import io.shiftleft.codepropertygraph.generated.nodes._
+import io.shiftleft.codepropertygraph.generated.nodes.*
 import io.shiftleft.codepropertygraph.generated.{DispatchTypes, EdgeTypes}
-import io.shiftleft.passes.CpgPass
-import io.shiftleft.semanticcpg.language._
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import io.shiftleft.semanticcpg.language.*
 import org.slf4j.{Logger, LoggerFactory}
 
-import scala.collection.mutable
+implicit val resolver: ICallResolver = NoResolve
+class StaticCallLinker(cpg: Cpg) extends ForkJoinParallelCpgPass[Seq[Method]](cpg) with LinkingUtil {
 
-class StaticCallLinker(cpg: Cpg) extends CpgPass(cpg) {
-
-  import StaticCallLinker._
-  private val methodFullNameToNode = mutable.Map.empty[String, List[Method]]
-
-  override def run(dstGraph: DiffGraphBuilder): Unit = {
+  private val logger: Logger = LoggerFactory.getLogger(classOf[StaticCallLinker])
 
-    cpg.method.foreach { method =>
-      methodFullNameToNode.updateWith(method.fullName) {
-        case Some(l) => Some(method :: l)
-        case None    => Some(List(method))
-      }
-    }
+  override def generateParts(): Array[Seq[Method]] = {
+    val allMethods = cpg.method.l
+    allMethods.grouped(getBatchSize(allMethods.size)).toArray
+  }
 
-    cpg.call.foreach { call =>
+  override def runOnPart(builder: DiffGraphBuilder, methods: Seq[Method]): Unit = {
+    methods.foreach { method =>
       try {
-        linkCall(call, dstGraph)
+        cpg.call
```

Comment:
In hindsight, we can already optimize this by filtering in the dispatch types directly.
```scala
cpg.call
   .dispatchType(DispatchTypes.STATIC_DISPATCH, DispatchTypes.INLINED)
   .methodFullNameExact
```